kernel/arch/dreamcast/hardware/maple/keyboard.c

```diff
@@ -643,7 +652,11 @@ static int kbd_attach(maple_driver_t *drv, maple_device_t *dev) {
         state->region = KBD_REGION_US;
 
     /* Make sure all the queue variables are set up properly... */
-    state->queue_tail = state->queue_head = state->queue_len = 0;
+    state->queue_tail = state->queue_head = 0;
+
+    const int irqs = irq_disable();
```

Comment:
This is only ever called during an irq callback. It gets called during autodetection here https://github.com/KallistiOS/KallistiOS/blob/1ff6f043444446a09d55e464fd9a73be5307c740/kernel/arch/dreamcast/hardware/maple/maple_irq.c#L131

Reply:
OH, SHIT. Thank you. I actually didn't realize that!